src/html.rs

```diff
@@ -95,88 +93,117 @@ pub fn walk_and_embed_assets(
                         }
                     }
 
+                    enum LinkType {
+                        Icon,
+                        Stylesheet,
+                        Preload,
+                        DnsPrefetch,
+                        Unknown,
+                    }
+
+                    let mut link_type = LinkType::Unknown;
                     for attr in attrs_mut.iter_mut() {
                         if &attr.name.local == "rel" {
-                            if is_icon(attr.value.as_ref()) {
-                                link_type = "icon";
+                            let value = attr.value.as_ref();
+                            if is_icon(value) {
+                                link_type = LinkType::Icon;
                                 break;
-                            } else if attr.value.as_ref() == "stylesheet" {
-                                link_type = "stylesheet";
+                            } else if value == "stylesheet" {
+                                link_type = LinkType::Stylesheet;
+                                break;
+                            } else if value == "preload" {
+                                link_type = LinkType::Preload;
+                                break;
+                            } else if value == "dns-prefetch" {
+                                link_type = LinkType::DnsPrefetch;
                                 break;
                             }
                         }
                     }
-
-                    if link_type == "icon" {
-                        for attr in attrs_mut.iter_mut() {
-                            if &attr.name.local == "href" {
-                                if opt_no_images {
-                                    attr.value.clear();
-                                } else {
-                                    let href_full_url =
-                                        resolve_url(&url, attr.value.as_ref()).unwrap_or_default();
-                                    let (favicon_dataurl, _) = retrieve_asset(
-                                        cache,
-                                        client,
-                                        &href_full_url,
-                                        true,
-                                        "",
-                                        opt_silent,
-                                    )
-                                    .unwrap_or_default();
-                                    attr.value.clear();
-                                    attr.value.push_slice(favicon_dataurl.as_str());
+                    let link_type = link_type;
```

Comment:
It's a shadowing idiom to make a variable immutable. Left hand side of of this line is immutable new variable and right hand side of this line is previous mutable variable.

It is better to rebind mutable variable as immutable variable because it is guaranteed not to be changed after this line.